src/components/MeasurementOption/MeasurementOption.js

```diff
@@ -35,59 +44,71 @@ class MeasurementOption extends React.PureComponent {
     openMeasurementDropdown: PropTypes.number,
   };
 
-  scaleFromRef = React.createRef();
-
-  scaleToRef = React.createRef();
+  onScaleChange = (value, type) => {
+    this.setState({ [type]: Number(value) }, () => {
+      this.props.onStyleChange('Scale', [
+        [this.state.currScaleFrom, this.state.currUnitFrom],
+        [this.state.currScaleTo, this.state.currUnitTo],
+      ]);
+    });
+  }
 
-  /**
-   * A callback function that is used to change the scale for a measurement tool
-   * The argument has the same format as this.props.scale
-   * If values in the array are falsy then current value will be used instead
-   */
-  onScaleChange = ([[scaleFrom, unitFrom], [scaleTo, unitTo]]) => {
-    const { scale } = this.props;
+  onPrecisionChange = (value, type) => {
+    this.setState({ [type]: Number(value) }, () => {
+      this.props.onStyleChange('Precision', this.state.currPrecision);
+    });
+  }
 
-    scaleFrom = scaleFrom || scale[0][0];
-    unitFrom = unitFrom || scale[0][1];
-    scaleTo = scaleTo || scale[1][0];
-    unitTo = unitTo || scale[1][1];
+  onUnitChange = (event, type) => {
+    this.setState({ [type]: event.target.value }, () => {
```

Comment:
I feel like we could combine `onUnitChange` together with `onScaleChange` as the setState callback is exactly the same. But I'm good with the code now, easy to understand.

Reply:
One is doing is converting a value to a number and the other one is doing string. Would have to write if check, but for now its easy to read and it is separated.

Reply:
Resolved in 084caf3. Thanks!